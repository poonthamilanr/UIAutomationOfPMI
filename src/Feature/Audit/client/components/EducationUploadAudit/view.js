import React from 'react';
import Document from 'foundation/Icon/client/document';
import Close from 'foundation/Icon/client/close';
import { Row, Col } from 'react-bootstrap';
import 'feature/Audit/client/components/AcademicEducationAudit/educationAudit.scss';
import { Button as BaseButton } from 'react-bootstrap';
import { LinkButton } from '@pmi/dsm-react-bs4';
import { connect } from 'react-redux';
import { Text } from '@sitecore-jss/sitecore-jss-react';
import * as auditAction from 'foundation/Application/client/AuditDocument/actions';

class AudiDocument extends React.PureComponent {
  handleDeleteClick = (auditDocument) => (e) => {
    e.preventDefault();
    e.disabled=true;
    this.props.deleteAudit(auditDocument);
  };

  handleDownloadClick = (auditDocument, updatedFileName) => (e) => {
    e.preventDefault();
    const param = {
      "audiDocument": auditDocument,
      "_links":auditDocument._links,
      "fileName": updatedFileName,
    }
    this.props.downlodAudit(param);
    return true;
  };

  render() {
    const { auditDocument, fileName, disabled, fields, isViewMode } = this.props;
    let updatedFileName = `${fileName} ${fields.FileFormatPostContent.value}`;
    if(auditDocument.documentMetadata !== null && auditDocument.documentMetadata.DocumentName !== undefined)
    {
      updatedFileName = auditDocument.documentMetadata.DocumentName;
    }
    if(isViewMode)
    {
      return (
        <Row className="docDownloadViewPanel">
          <Col md={12}>
            {updatedFileName}
            <BaseButton as="a" variant="link" href={auditDocument.url} download={updatedFileName} onClick={auditDocument.url === "" || auditDocument.url === undefined ? this.handleDownloadClick(auditDocument, updatedFileName) : ""} className="view-doc-link btn-md link-base"><Text field={fields.FileDownloadLink} /></BaseButton>
          </Col>
        </Row>
      )
    }
    return (
      <Row className="docDownloadPanel">
        <Col md={11}><BaseButton as="a" variant="link" href={auditDocument.url} download={updatedFileName} onClick={auditDocument.url === "" || auditDocument.url === undefined ? this.handleDownloadClick(auditDocument, updatedFileName) : ""} className="with-icon download_link btn-md link-base"><Document /> {updatedFileName}</BaseButton></Col>
        <Col md={1}>
          <LinkButton
            className="link-base remove_document"
            onClick={this.handleDeleteClick(auditDocument)}
            icon={Close}
            disabled={disabled}
          />
        </Col>
      </Row>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  deleteAudit: (data) => dispatch(auditAction.deleteAuditDocument(data)),
  downlodAudit: (data) => dispatch(auditAction.downloadAuditDocument(data)),
});

export default connect(null, mapDispatchToProps)(AudiDocument);
