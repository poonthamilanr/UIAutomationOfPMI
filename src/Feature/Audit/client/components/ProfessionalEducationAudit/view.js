import React from 'react';
import 'feature/Audit/client/components/AcademicEducationAudit/educationAudit.scss';
import AuditDocumentView from 'feature/Audit/client/components/EducationUploadAudit/view';
import EducationUploadAuditModal from 'feature/Audit/client/components/EducationUploadAudit';
import { Row, Col, Button } from 'react-bootstrap';
import { ReactComponent as DownloadIcon } from '../../../../../Project/Certification/client/assets/icons/upload.svg';

class ProfessionalEducationAuditView extends React.Component {
  state = {
    isOpen: false,
  };

  auditDocumentView(auditDocument, fields, fileName, isViewMode) {
    return (
      <AuditDocumentView
        key={auditDocument._links.self.href}
        auditDocument={auditDocument}
        fields={fields.UploadModal.fields}
        fileName={fileName}
        isViewMode={isViewMode}
      />
    );
  }

  handleShow = () => {
    this.setState({ isOpen: true });
  };

  handleClose = () => {
    this.setState({ isOpen: false });
  };

  getDisplatDate(value) {
    const { sitecoreListsSettings } = this.props;
    const date = new Date(value);
    const month = value ? sitecoreListsSettings.Months[date.getMonth()].displayName.substring(0, 3) : '';
    const year = date.getFullYear();
    return `${month} ${year}`;
  }

  getDisplayDatePeriod(dateFrom, dateTo) {
    if (!dateFrom || !dateTo) {
      return null;
    }

    return `${this.getDisplatDate(dateFrom)} - ${this.getDisplatDate(dateTo)}`;
  }

  render() {
    const { fields, education, auditDocument, isViewMode } = this.props;
    const { isOpen } = this.state;
    if (!fields) return <div className="static-height" />;
    if (auditDocument === null || auditDocument === undefined) {
      return null;
    }
    const isUploaded = auditDocument.documentReceived;
    return (
      <>
        <div className="educationInfoBox mb-5">
          <Row>
            <Col sm={9}>
              <span className="fontBold">{education.courseTitle} </span>
              <br />
              <span>{education.institution}</span> <br />
              <span>
                {' '}
                {this.getDisplayDatePeriod(education.courseStartDate, education.courseEndDate)} | {education.hoursTotal}{' '}
                Hours
              </span>
              <br />
            </Col>
            <Col sm={3}>
              {!isUploaded && !isViewMode ? (
                <div className="dsm d-flex justify-content-end p-0">
                  <Button
                    as="a"
                    variant="link"
                    onClick={this.handleShow}
                    className="btn btn-link btn-md with-icon uploadBtn"
                  >
                    <div className="btn-icon uploadIcon">
                      <DownloadIcon />
                    </div>
                    <div>{fields.UploadLabel.value}</div>
                  </Button>
                </div>
              ) : (
                ''
              )}
            </Col>
          </Row>
          {isUploaded ? this.auditDocumentView(auditDocument, fields, education.institution, isViewMode) : ''}
        </div>
        <EducationUploadAuditModal
          show={isOpen}
          onHide={this.handleClose}
          auditDocument={auditDocument}
          fields={fields.UploadModal.fields}
        />
      </>
    );
  }
}

export default ProfessionalEducationAuditView;
