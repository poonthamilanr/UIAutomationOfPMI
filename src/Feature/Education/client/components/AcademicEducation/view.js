import React from 'react';
import { connect } from 'react-redux';
import { LinkButton } from '@pmi/dsm-react-bs4';
import { Text } from '@sitecore-jss/sitecore-jss-react';
import i18n from 'i18next';
import classNames from 'classnames';
import IconPencil from '@pmi/dsm-react-bs4/dist/components/icons/16/PencilIcon';
import { getEnumDisplayName } from 'foundation/SitecoreSettings/client/utils';
import * as uiActions from "./actions";
import './academicEducationView.scss';

const Property = ({ value, enabled, label }) => {
  if (!value || (enabled !== undefined && !enabled)) {
    return null;
  }

  return (
    <div className="academic-education-property">
      <div className="academic-education-property__label">{label}</div>
      <div className="academic-education-property__value">{value}</div>
    </div>
  );
};

const AcademicEducationView = (props) => {
  const { academicEducation, openForm, hideChangeOption, settings, className, fields } = props;

  if (!academicEducation) {
    return null;
  }

  const {
    degreeEnum,
    schoolName,
    fieldOfStudyEnum,
    yearStarted,
    yearOfDegree,
    schoolCountry,
    accreditedUniversityDegree,
  } = academicEducation;

  const isUniversity = degreeEnum && degreeEnum.toLowerCase() !== 'highschooldiploma';
  const isDegreeProgram = academicEducation.accreditedUniversityDegreeId;
  const requirementsMet = true;

  return (
    <div className={classNames(className, "academic-education", "academic-education__view")}>
      <div className="academic-education__header d-flex flex-row">
        <span>{getEnumDisplayName(settings.academicEducationLevels, degreeEnum)}</span>
        {!requirementsMet &&
          <>
            <span className="ml-1 mr-1"> - </span>
            <span className="error"><Text field={fields.ViewModeRequirementsNotMet} /></span>
          </>}
      </div>
      <div className="academic-education__body">
        <div className="academic-education__properies-column">
          <Property
            value={yearOfDegree && [yearStarted, yearOfDegree].filter(v => v).join(' - ')}
            label={<Text field={fields.ViewModeYearsAttended} />}
          />
          <Property
            value={schoolName}
            label={<Text field={isUniversity ? fields.ViewModeNameOfInstitution : fields.ViewModeNameOfSchool} />}
          />
        </div>
        <div className="academic-education__properies-column">
          <Property
            value={schoolCountry && schoolCountry.country}
            label={<Text field={fields.ViewModeCountryOfInstitution} />}
          />
          <Property
            value={isDegreeProgram ?
              accreditedUniversityDegree && accreditedUniversityDegree.name :
              getEnumDisplayName(settings.fieldOfStudy, fieldOfStudyEnum)}
            label={<Text field={isDegreeProgram ? fields.ViewModeDegreeProgram : fields.ViewModeFieldOfStudy} />}
          />
        </div>
      </div>
      {!hideChangeOption && (
        <div className="academic-education__buttons">
          <LinkButton
            className="with-icon link-base"
            onClick={openForm}
            icon={IconPencil}
            titleText={i18n.t('cert-app.Education.AcademicEducation.EditEducation')}
          />
        </div>
      )}
    </div>
  );
}

const mapDispatchToProps = dispatch => ({
  openForm: () => dispatch(uiActions.setIsOpen(true)),
});

export default connect(null, mapDispatchToProps)(AcademicEducationView);
