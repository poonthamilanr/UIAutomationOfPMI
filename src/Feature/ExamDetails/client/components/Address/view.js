import React from 'react';
import { connect } from 'react-redux';
import { Button, LinkButton } from '@pmi/dsm-react-bs4';
import IconPencil from '@pmi/dsm-react-bs4/dist/components/icons/16/PencilIcon';
import { Text } from '@sitecore-jss/sitecore-jss-react';
import { getCountriesInfo, getAllCountries, getFilteredCountries } from 'foundation/Profile/client/Metadata/countries';
import { getLocationStatus } from 'feature/ExamDetails/client/components/ExamDetails/accessors';
import i18n from 'i18next';
import * as pageActions from "../ExamDetails/actions";

class AddressView extends React.Component {
  getDisplayAddressType(address) {
    const { fields } = this.props;
    const addressType = address && address.AddressType;

    switch (addressType) {
    case 'Home': return fields.AddressTypeHomeAddress.value;
    case 'Work': return fields.AddressTypeWorkAddress.value;
    default : return addressType ? `${addressType} ${fields.AddressTypeAddress.value}` : fields.AddressTypeAddress.value;
    }
  }

  render() {
    const { openForm, address, countriesData, fields, isValidExamLocation } = this.props;
    const countries = countriesData || [];
    let country = '';
    if (address) {
      const conuntryObj = countries.find(data => data.countryCode3 === address.countryCode) || {};
      country = conuntryObj.country || address.countryCode;
    }
    const cssClass = isValidExamLocation ? "view-mode" : "view-mode disable-view";
    return (
      <div className={cssClass}>
        <div className="view-mode__exam">
          <h2><Text field={fields.ViewModeTitle} /></h2>
          {address && (
            <address>
              <p><b>{this.getDisplayAddressType(address)}</b></p>
              <p>{address.address1}</p>
              {address.address2 && <p>{address.address2}</p>}
              <p>{address.city}, {address.state} {address.postalCode}</p>
              <p>{country}</p>
            </address>
          )}
        </div>
        {address && (
          <>
            <LinkButton
              className="btn with-icon link-base d-none d-md-flex"
              onClick={openForm}
              icon={IconPencil}
              titleText={i18n.t('cert-app.ExamDetails.Address.EditAddressLabel')}
            />
            <Button
              variant="outline-primary"
              className="d-md-none mt-2"
              onClick={openForm}
              titleText={i18n.t('cert-app.ExamDetails.Address.EditAddressLabel')}
            />
          </>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  countriesData: getAllCountries(getCountriesInfo(state)) || getFilteredCountries(getCountriesInfo(state)),
  isValidExamLocation: getLocationStatus(state),
})

const mapDispatchToProps = dispatch => ({
  openForm: () => dispatch(pageActions.openForm('address')),
});

const addressView = connect(
  mapStateToProps,
  mapDispatchToProps,
)(AddressView);

export default addressView;
