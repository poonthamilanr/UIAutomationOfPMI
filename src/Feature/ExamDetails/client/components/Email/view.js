import React from 'react';
import { connect } from 'react-redux';
import { Button, LinkButton } from '@pmi/dsm-react-bs4';
import IconPencil from '@pmi/dsm-react-bs4/dist/components/icons/16/PencilIcon';
import { Text } from '@sitecore-jss/sitecore-jss-react';
import i18n from 'i18next';
import { getLocationStatus } from 'feature/ExamDetails/client/components/ExamDetails/accessors';
import * as pageActions from "../ExamDetails/actions";

export class EmailViewBase extends React.PureComponent {
  render() {
    const { email, openForm, fields, isValidExamLocation  } = this.props;
    const cssClass = isValidExamLocation ? "view-mode" : "view-mode disable-view";
    return (
      <div className={cssClass}>
        <div className="view-mode__exam">
          <h2><Text field={fields.ViewModeTitle} /></h2>
          {email && <p>{email}</p>}
        </div>
        {email && (
          <>
            <LinkButton
              variant="link"
              className="with-icon link-base d-none d-md-flex"
              onClick={openForm}
              icon={IconPencil}
              titleText={i18n.t('cert-app.ExamDetails.Email.EditEmailLabel')}
            />
            <Button
              variant="outline-primary"
              className="d-md-none mt-2"
              onClick={openForm}
              titleText={i18n.t('cert-app.ExamDetails.Email.EditEmailLabel')}
            />
          </>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isValidExamLocation: getLocationStatus(state),
});

const mapDispatchToProps = dispatch => ({
  openForm: () => dispatch(pageActions.openForm('email')),
});

const emailView = connect(
  mapStateToProps,
  mapDispatchToProps,
)(EmailViewBase);

export default emailView;
