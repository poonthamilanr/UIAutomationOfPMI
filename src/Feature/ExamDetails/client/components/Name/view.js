import React from 'react';
import { connect } from 'react-redux';
import { Button, LinkButton } from '@pmi/dsm-react-bs4';
import IconPencil from '@pmi/dsm-react-bs4/dist/components/icons/16/PencilIcon';
import { Text } from '@sitecore-jss/sitecore-jss-react';
import { getLocationStatus } from 'feature/ExamDetails/client/components/ExamDetails/accessors';
import i18n from 'i18next';
import * as pageActions from "../ExamDetails/actions";

class NameView extends React.PureComponent {
  render() {
    const { nameOnIdentification, openForm, fields, isValidExamLocation } = this.props;
    const cssClass = isValidExamLocation ? "view-mode" : "view-mode disable-view";
    return (
      <div className={cssClass}>
        <div className="view-mode__exam">
          <h2><Text field={fields.ViewModeNameOnIdentificationTitle} /></h2>
          {nameOnIdentification &&
            <p>{nameOnIdentification.firstName} {nameOnIdentification.middleName} {nameOnIdentification.lastName}</p>}
        </div>
        {nameOnIdentification && (
          <>
            <LinkButton
              className="btn with-icon link-base d-none d-md-flex"
              onClick={openForm}
              icon={IconPencil}
              titleText={i18n.t('cert-app.ExamDetails.Name.EditName')}
            />
            <Button
              variant="outline-primary"
              className="d-md-none mt-2"
              onClick={openForm}
              titleText={i18n.t('cert-app.ExamDetails.Name.EditName')}
            />
          </>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isValidExamLocation: getLocationStatus(state),
});

const mapDispatchToProps = dispatch => ({
  openForm: () => dispatch(pageActions.openForm('name')),
});

export default connect(mapStateToProps, mapDispatchToProps)(NameView);
