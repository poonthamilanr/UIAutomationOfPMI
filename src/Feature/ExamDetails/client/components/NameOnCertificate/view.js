import React from 'react';
import { connect } from 'react-redux';
import { Button, LinkButton } from '@pmi/dsm-react-bs4';
import IconPencil from '@pmi/dsm-react-bs4/dist/components/icons/16/PencilIcon';
import { Text } from '@sitecore-jss/sitecore-jss-react';
import { getLocationStatus } from 'feature/ExamDetails/client/components/ExamDetails/accessors';
import i18n from 'i18next';
import * as pageActions from "../ExamDetails/actions";

class NameOnCertificateView extends React.PureComponent {
  render() {
    const { nameOnCertificate, openForm, fields, isValidExamLocation } = this.props;
    const cssClass = isValidExamLocation ? "view-mode" : "view-mode disable-view";
    return (
      <div className={cssClass}>
        <div className="view-mode__exam">
          <h2><Text field={fields.ViewModeNameOnCertificateTitle} /></h2>
          {nameOnCertificate &&
            <p>{nameOnCertificate}</p>}
        </div>
        {nameOnCertificate && (
          <>
            <LinkButton
              className="btn with-icon link-base d-none d-md-flex"
              onClick={openForm}
              icon={IconPencil}
              titleText={i18n.t('cert-app.ExamDetails.NameOnCertificate.EditName')}
            />
            <Button
              variant="outline-primary"
              className="d-md-none mt-2"
              onClick={openForm}
              titleText={i18n.t('cert-app.ExamDetails.NameOnCertificate.EditName')}
            />
          </>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isValidExamLocation: getLocationStatus(state),
});

const mapDispatchToProps = dispatch => ({
  openForm: () => dispatch(pageActions.openForm('nameOnCertificate')),
});

export default connect(mapStateToProps, mapDispatchToProps)(NameOnCertificateView);
