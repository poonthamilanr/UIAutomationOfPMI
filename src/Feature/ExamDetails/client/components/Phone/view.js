import React from 'react';
import { connect } from 'react-redux';
import { Button, LinkButton } from '@pmi/dsm-react-bs4';
import IconPencil from '@pmi/dsm-react-bs4/dist/components/icons/16/PencilIcon';
import { Text } from '@sitecore-jss/sitecore-jss-react';
import { getLocationStatus } from 'feature/ExamDetails/client/components/ExamDetails/accessors';
import i18n from 'i18next';
import * as pageActions from "../ExamDetails/actions";

class PhoneView extends React.Component {

  getDisplayPhoneNumber(phone) {
    const countryCode = phone.phoneCountryCode || '';
    const areaCode = phone.phoneAreaCode || '';
    const number = phone.phoneNumber || '';
    return countryCode && (areaCode || number) ? `+${countryCode} ${areaCode}${number}` : '';
  }

  getDisplayPhoneExtension(phone) {
    return phone.phoneExtension || '';
  }

  getDisplayPhoneType(phone) {
    const { fields } = this.props;
    const phoneType = phone.PhoneNumberType;

    switch (phoneType) {
    case 'Cell': return fields.PhoneNumberTypeCell.value;
    case 'Home': return fields.PhoneNumberTypeHome.value;
    case 'Work': return fields.PhoneNumberTypeWork.value;
    default : return phoneType ? `${phoneType}` : '';
    }
  }

  render() {
    const { phone, openForm, fields, isValidExamLocation } = this.props;
    const phoneNumber = phone && this.getDisplayPhoneNumber(phone);
    const phoneExtension = phone && this.getDisplayPhoneExtension(phone);
    const phoneType = phone && this.getDisplayPhoneType(phone).toLowerCase();
    const hasValue = val => val;
    const cssClass = isValidExamLocation ? "view-mode" : "view-mode disable-view";

    return (
      <div className={cssClass}>
        <div className="view-mode__exam">
          <h2><Text field={fields.ViewModeTitle} /></h2>
          {[
            phoneNumber,
            phoneExtension && `${fields.ViewModeExtensionOptionalKey.value}:${phoneExtension}`,
            phoneType && `(${phoneType})`,
          ].filter(hasValue).join(' ')}
        </div>
        {phone.phoneNumber && (
          <>
            <LinkButton
              className="with-icon link-base d-none d-md-flex"
              onClick={openForm}
              icon={IconPencil}
              titleText={i18n.t('cert-app.ExamDetails.Phone.EditPhone')}
            />
            <Button
              variant="outline-primary"
              className="d-md-none mt-2"
              onClick={openForm}
              titleText={i18n.t('cert-app.ExamDetails.Phone.EditPhone')}
            />
          </>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isValidExamLocation: getLocationStatus(state),
});

const mapDispatchToProps = dispatch => ({
  openForm: () => dispatch(pageActions.openForm('phone')),
});

const phoneView = connect(
  mapStateToProps,
  mapDispatchToProps,
)(PhoneView);

export default phoneView;
