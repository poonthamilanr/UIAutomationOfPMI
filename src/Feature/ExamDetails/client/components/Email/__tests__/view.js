import React from 'react';
// import { shallow } from 'enzyme';
import { EmailViewBase } from '../view';

describe('EmailView', () => {
  it('should render correctly with email', () => {
    // const component = shallow(<EmailViewBase email="some@email.com" />);
    const component = <EmailViewBase email="some@email.com" />;

    expect(component).toMatchSnapshot();
  });

  it('should render correctly without email', () => {
    // const component = shallow(<EmailViewBase email="" />);
    const component = <EmailViewBase email="" />;

    expect(component).toMatchSnapshot();
  });
});