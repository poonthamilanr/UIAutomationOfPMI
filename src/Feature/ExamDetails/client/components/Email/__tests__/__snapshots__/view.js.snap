// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailView should render correctly with email 1`] = `
<EmailViewBase
  email="some@email.com"
/>
`;

exports[`EmailView should render correctly without email 1`] = `
<EmailViewBase
  email=""
/>
`;
